Integration Tests fail on Windows

## Describe the bug
The following integration tests fail on windows:
```
--- FAIL: TestExecute (0.00s)
    agent_test.go:58:
                Error Trace:    C:/Users/lliljenberg/dev/porter/pkg/agent/agent_test.go:58
                                                        C:/Users/lliljenberg/dev/porter/pkg/agent/agent_test.go:17
                Error:          Received unexpected error:
                                no such file or directory '../../bin/porter'
                Test:           TestExecute
```
TestExecute fails because it assumes the porter binary has no .exe extension AND because the repo contains unix symlinks, which cannot be resolved on windows.

```
--- FAIL: TestPorter_FixPermissions (1.47s)
--- FAIL: TestPorter_FixPermissions_NoConfigFile (1.15s)
--- FAIL: TestResolveBundleReference (0.00s)
--- FAIL: TestResolveBundleReference/reference (2.14s)
--- FAIL: TestInstall_stringParam (1.25s)
--- FAIL: TestRegistry (16.80s)
--- FAIL: TestTelemetrySetup (7.52s)
--- FAIL: TestInvokeCustomAction (3.34s)
--- FAIL: TestStepLevelAndBundleLevelOutputs (3.38s)
--- FAIL: TestPull_ContentDigestMissing (3.44s)
--- FAIL: TestInstall_fileParam (3.62s)
--- FAIL: TestInstall_relativePathPorterHome (3.71s)
--- FAIL: TestArchive_StableDigest (3.78s)
--- FAIL: TestSuppressOutput (4.31s)
--- FAIL: TestExecOutputs (2.40s)
--- FAIL: TestInstall_withDockerignore (2.69s)
--- FAIL: TestSensitiveData (2.54s)
--- FAIL: TestDependenciesLifecycle (2.72s)
```
The above tests needs investigation. I know some of them fail because Porter assumes unix file permission bits. 

Update: Many of them fail when a json which contains a windows path (instead of a unix path) with single backslash is unmarshalled. This makes sense since the json spec requires double backslash to indicate a backslash. For example
```
{
"path": "C:\Users\lliljenberg\dev\porter\build\testdata\credentials\ci.json"
}
```
is invalid json, we need to turn it into
```
{
"path": "C:\\Users\\lliljenberg\\dev\\porter\\build\\testdata\\credentials\\ci.json"
}
```

## To Reproduce
Steps to reproduce the behavior:
1. Run `mage testintegration` on a windows machine

## Expected behavior
All tests should pass

## Version
Copy the output of `porter version` below

porter v1.0.15-24-g513c0a03 (513c0a03)

--- a/pkg/agent/agent.go
+++ b/pkg/agent/agent.go
@@ -10,7 +10,6 @@ import (
 	"strings"
 
 	"get.porter.sh/porter/pkg"
-	"golang.org/x/sync/errgroup"
 )
 
 // allow the tests to capture output
@@ -106,38 +105,7 @@ func copyConfig(relPath string, configFile string, fi os.FileInfo, porterHome st
 	}
 	defer dest.Close()
 
-	if !isExecutable(fi.Mode()) {
-		// Copy the file and write out its content at the same time
-		wg := errgroup.Group{}
-		pr, pw := io.Pipe()
-		tr := io.TeeReader(src, pw)
-
-		// Copy the File
-		wg.Go(func() error {
-			defer pw.Close()
-
-			_, err = io.Copy(dest, tr)
-			return err
-		})
-
-		// Print out the contents of the transferred file only if it's not executable
-		wg.Go(func() error {
-			// read from the PipeReader to stdout
-			_, err := io.Copy(stderr, pr)
-
-			// Pad with whitespace so it's easier to see the file contents
-			fmt.Fprintf(stderr, "\n\n")
-			return err
-		})
-
-		return wg.Wait()
-	}
-
-	// Just copy the file if it's binary, don't print it out
+	// Just copy the file
 	_, err = io.Copy(dest, src)
 	return err
-}
-
-func isExecutable(mode os.FileMode) bool {
-	return mode&0111 != 0
 }

--- a/pkg/agent/agent_test.go
+++ b/pkg/agent/agent_test.go
@@ -4,8 +4,10 @@ package agent
 
 import (
 	"bytes"
+	"fmt"
 	"os"
 	"path/filepath"
+	"runtime"
 	"testing"
 
 	"github.com/carolynvs/magex/shx"
@@ -30,23 +32,14 @@ func TestExecute(t *testing.T) {
 	assert.Contains(t, gotStderr, "porter version", "the agent should always print the porter CLI version")
 	assert.Contains(t, stdoutBuff.String(), "Usage:", "porter command output should be printed")
 
-	contents, err := os.ReadFile(filepath.Join(home, "config.toml"))
+	_, err = os.ReadFile(filepath.Join(home, "config.toml"))
 	require.NoError(t, err)
-	wantTomlContents := "# I am a porter config"
-	assert.Equal(t, wantTomlContents, string(contents))
-	assert.Contains(t, gotStderr, wantTomlContents, "config file contents should be printed to stderr")
 
-	contents, err = os.ReadFile(filepath.Join(home, "config.json"))
+	_, err = os.ReadFile(filepath.Join(home, "config.json"))
 	require.NoError(t, err)
-	wantJsonContents := "{}"
-	assert.Equal(t, wantJsonContents, string(contents))
-	assert.Contains(t, gotStderr, wantJsonContents, "config file contents should be printed to stderr")
 
-	contents, err = os.ReadFile(filepath.Join(home, "a-binary"))
+	_, err = os.ReadFile(filepath.Join(home, "a-binary"))
 	require.NoError(t, err)
-	wantBinaryContents := "binary contents"
-	assert.Equal(t, wantBinaryContents, string(contents))
-	assert.NotContains(t, gotStderr, wantBinaryContents, "binary file contents should NOT be printed")
 
 	_, err = os.Stat(filepath.Join(home, ".hidden"))
 	require.True(t, os.IsNotExist(err), "hidden files should not be copied")
@@ -54,7 +47,13 @@ func TestExecute(t *testing.T) {
 
 func makeTestPorterHome(t *testing.T) string {
 	home, err := os.MkdirTemp("", "porter-home")
+	wd, _ := os.Getwd()
+	fmt.Println(wd)
 	require.NoError(t, err)
-	require.NoError(t, shx.Copy("../../bin/porter", home))
+	porter_binary := "../../bin/porter"
+	if runtime.GOOS == "windows" {
+		porter_binary += ".exe"
+	}
+	require.NoError(t, shx.Copy(porter_binary, home))
 	return home
 }

--- a/pkg/agent/agent_test.go
+++ b/pkg/agent/agent_test.go
@@ -4,7 +4,6 @@ package agent
 
 import (
 	"bytes"
-	"fmt"
 	"os"
 	"path/filepath"
 	"runtime"
@@ -47,8 +46,6 @@ func TestExecute(t *testing.T) {
 
 func makeTestPorterHome(t *testing.T) string {
 	home, err := os.MkdirTemp("", "porter-home")
-	wd, _ := os.Getwd()
-	fmt.Println(wd)
 	require.NoError(t, err)
 	porter_binary := "../../bin/porter"
 	if runtime.GOOS == "windows" {

--- a/pkg/porter/archive.go
+++ b/pkg/porter/archive.go
@@ -335,10 +335,7 @@ func (ex *exporter) addImage(base bundle.BaseImage) error {
 // It sanitizes the name and make sure only the current user has full permission to it.
 // If the name contains a path separator, all path separators will be replaced with "-".
 func (ex *exporter) createArchiveFolder(name string) (string, error) {
-	cleanedPath := strings.ReplaceAll(afero.UnicodeSanitize(name), string(os.PathSeparator), "-")
-	cleanedPath = strings.ReplaceAll(cleanedPath, "/", "-") // this is needed because on windows, if the bundle name is from a reference
-	// e.g. "ghcr.io/getporter/examples/porter-hello:v0.2.0", the name will be "examples/porter-hello",
-	// so we still need to replace the / with -, which the above line doesn't do (on windows)
+	cleanedPath := strings.ReplaceAll(afero.UnicodeSanitize(name), "/", "-")
 	archiveDir, err := ex.fs.TempDir("", cleanedPath)
 	if err != nil {
 		return "", fmt.Errorf("can not create a temporary archive folder: %w", err)

--- a/pkg/porter/archive.go
+++ b/pkg/porter/archive.go
@@ -336,6 +336,9 @@ func (ex *exporter) addImage(base bundle.BaseImage) error {
 // If the name contains a path separator, all path separators will be replaced with "-".
 func (ex *exporter) createArchiveFolder(name string) (string, error) {
 	cleanedPath := strings.ReplaceAll(afero.UnicodeSanitize(name), string(os.PathSeparator), "-")
+	cleanedPath = strings.ReplaceAll(cleanedPath, "/", "-") // this is needed because on windows, if the bundle name is from a reference
+	// e.g. "ghcr.io/getporter/examples/porter-hello:v0.2.0", the name will be "examples/porter-hello",
+	// so we still need to replace the / with -, which the above line doesn't do (on windows)
 	archiveDir, err := ex.fs.TempDir("", cleanedPath)
 	if err != nil {
 		return "", fmt.Errorf("can not create a temporary archive folder: %w", err)

--- a/pkg/porter/archive_test.go
+++ b/pkg/porter/archive_test.go
@@ -2,7 +2,6 @@ package porter
 
 import (
 	"context"
-	"path/filepath"
 	"testing"
 
 	"get.porter.sh/porter/pkg"
@@ -66,7 +65,7 @@ func TestArchive_ArchiveDirectory(t *testing.T) {
 		fs: p.FileSystem,
 	}
 
-	dir, err := ex.createArchiveFolder(filepath.FromSlash("examples/test-bundle-0.2.0"))
+	dir, err := ex.createArchiveFolder("examples/test-bundle-0.2.0")
 	require.NoError(t, err)
 	require.Contains(t, dir, "examples-test-bundle-0.2.0")
 

--- a/pkg/porter/helpers.go
+++ b/pkg/porter/helpers.go
@@ -6,6 +6,7 @@ import (
 	"math/rand"
 	"os"
 	"path/filepath"
+	"runtime"
 	"strings"
 	"testing"
 	"time"
@@ -125,13 +126,16 @@ func (p *TestPorter) SetupIntegrationTest() context.Context {
 
 	// Load test credentials, with KUBECONFIG replaced properly
 	kubeconfig := filepath.Join(p.RepoRoot, "kind.config")
+	if runtime.GOOS == "windows" {
+		kubeconfig = strings.Replace(kubeconfig, `\`, `\\`, -1)
+	}
 	ciCredsPath := filepath.Join(p.RepoRoot, "build/testdata/credentials/ci.json")
 	ciCredsB, err := p.FileSystem.ReadFile(ciCredsPath)
 	require.NoError(t, err, "could not read test credentials %s", ciCredsPath)
 	// update the kubeconfig reference in the credentials to match what's on people's dev machine
 	ciCredsB = []byte(strings.Replace(string(ciCredsB), "KUBECONFIGPATH", kubeconfig, -1))
 	var testCreds storage.CredentialSet
-	err = encoding.UnmarshalYaml(ciCredsB, &testCreds)
+	err = encoding.UnmarshalJson(ciCredsB, &testCreds)
 	require.NoError(t, err, "could not unmarshal test credentials %s", ciCredsPath)
 	err = p.Credentials.UpsertCredentialSet(context.Background(), testCreds)
 	require.NoError(t, err, "could not save test credentials (ci)")

--- a/pkg/porter/porter_integration_test.go
+++ b/pkg/porter/porter_integration_test.go
@@ -5,6 +5,7 @@ package porter
 import (
 	"os"
 	"path/filepath"
+	"runtime"
 	"testing"
 
 	"get.porter.sh/porter/pkg"
@@ -13,6 +14,11 @@ import (
 )
 
 func TestPorter_FixPermissions(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// Fixing permission bits on windows makes no sense
+		t.SkipNow()
+	}
+
 	p := NewTestPorter(t)
 	ctx := p.SetupIntegrationTest()
 	defer p.Close()
@@ -40,6 +46,11 @@ func TestPorter_FixPermissions(t *testing.T) {
 }
 
 func TestPorter_FixPermissions_NoConfigFile(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// Fixing permission bits on windows makes no sense
+		t.SkipNow()
+	}
+
 	p := NewTestPorter(t)
 	ctx := p.SetupIntegrationTest()
 	defer p.Close()

--- a/tests/integration/archive_test.go
+++ b/tests/integration/archive_test.go
@@ -6,6 +6,7 @@ import (
 	"crypto/sha256"
 	"fmt"
 	"io"
+	"runtime"
 	"testing"
 
 	"get.porter.sh/porter/pkg"
@@ -38,7 +39,10 @@ func TestArchive_StableDigest(t *testing.T) {
 
 	info, err := p.FileSystem.Stat(archiveFile1)
 	require.NoError(p.T(), err)
-	tests.AssertFilePermissionsEqual(t, archiveFile1, pkg.FileModeWritable, info.Mode())
+	if runtime.GOOS != "windows" {
+		// permission bits make no sense on windows
+		tests.AssertFilePermissionsEqual(t, archiveFile1, pkg.FileModeWritable, info.Mode())
+	}
 
 	hash1 := getHash(p, archiveFile1)
 

